addons/stock/models/stock_picking.py

```diff
@@ -504,6 +504,17 @@ def create(self, vals):
 
     @api.multi
     def write(self, vals):
+        # Log an activity for following picking indicating that initial demand has been lowered
+        if vals.get('move_lines'):
+            self.ensure_one()
+            log_moves = {}
+            for code, move_id, move_values in vals.get('move_lines'):
+                move = self.env['stock.move'].browse(move_id)
```

Comment:
i'm not sure this is the good place ; you'd have to handle all the x2many magic code.. isn't it possible to move the logic in the write of move? or you could check before and after the call to super here